infer/src/base/ProcessPool.ml

```diff
@@ -179,14 +179,18 @@ let wait_for_updates pool buffer =
   update
 
 
-let killall pool ~slot status =
-  Array.iter pool.slots ~f:(fun {pid} ->
+let killall_silently slots =
+  (* Note: do not write anything in this function, as it may execute after the file
+     descriptors to stdout/stderr have been closed. *)
+  Array.iter slots ~f:(fun {pid} ->
       match Signal.send Signal.term (`Pid pid) with `Ok | `No_such_process -> () ) ;
-  Array.iter pool.slots ~f:(fun {pid} ->
+  Array.iter slots ~f:(fun {pid} ->
       try Unix.wait (`Pid pid) |> ignore
-      with Unix.Unix_error (ECHILD, _, _) ->
-        (* some children may have died already, it's fine *) () ) ;
-  ProcessPoolState.has_running_children := false ;
+      with Unix.Unix_error _ -> (* some children may have died already, it's fine *) () )
```

Comment:
Was matching ECHILD too restrictive?

Reply:
IIRC, we had one case where the error code was not the same on Windows, but this was related to files. Will experiment too.